common-service-core/src/test/java/com/ca/mfaas/util/FileUtilsTest.java

```diff
@@ -0,0 +1,164 @@
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.util;
+
+import org.junit.*;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.InvalidPathException;
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertNotNull;
+
+
+@RunWith(JUnit4.class)
+public class FileUtilsTest {
+    private static final String RELATIVE_PATH_NAME = "Documents/apiml";
+    private static final String CONFIG_PATH = System.getProperty("user.home").replace("\\", "/") + File.separator + RELATIVE_PATH_NAME;
+    private static boolean folderCreated = false;
+
+    private static String fileName = "service-configuration.yml";
+    private static File configFile;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @BeforeClass
+    public static void setUp() {
+        File customDir = new File(CONFIG_PATH);
+        if (!customDir.exists()) {
+            customDir.mkdirs();
+            folderCreated = true;
+        }
+
+        if ((configFile == null) || !configFile.exists()) {
+            configFile = new File(CONFIG_PATH + File.separator + fileName);
+            try {
+                configFile.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+    @AfterClass
+    public static void cleanUp() {
+        if ((configFile != null) && configFile.exists()) {
+            configFile.delete();
+        }
+
+        if (folderCreated) {
+            File documentsFolder = new File(CONFIG_PATH);
+            documentsFolder.delete();
+        }
+
+    }
+
+    @Test
+    public void testLocateFileForNull() {
+        File aFile = FileUtils.locateFile(null);
+        assertNull(aFile);
+    }
+
+    /**
+     * Locate file as resource or a System resource
+     */
+    @Test
+    public void testLocateFileAsResource() {
+
+        // Resource accessible from this class classloader
+        String fileName = "/service-configuration.yml";
+
+        File aFile = FileUtils.locateFile(fileName);
+        assertNotNull(aFile);
+        assertEquals("service-configuration.yml", aFile.getName());
+        assertNotNull(aFile.canRead());
+
+        // Resource accessible using System classloader
+        fileName = "service-configuration.yml";
+        aFile = FileUtils.locateFile(fileName);
+        assertNotNull(aFile);
+        assertEquals("service-configuration.yml", aFile.getName());
+        assertNotNull(aFile.canRead());
+    }
+    @Test
+    public void testLocateFileAsDirectory() {
+        String fileName = "/";
+
+        File aFile = FileUtils.locateDirectory(fileName);
+        assertNotNull(aFile);
+        assertNotNull(aFile.canRead());
```

Comment:
aFile.canRead() returns value in primitive type. It will be always not null after casting. Please make sure about your assertion.